src/libstd/sys/unix/sync.rs

```diff
@@ -249,3 +249,59 @@ mod os {
     };
     pub const PTHREAD_MUTEX_RECURSIVE: libc::c_int = 1;
 }
+
+#[cfg(target_os = "netbsd")]
+mod os {
+    use libc;
+
+    // size of the type minus width of the magic int field
+    #[cfg(target_arch = "x86_64")]
+    const __PTHREAD_MUTEX_SIZE__: usize = 48 - 4;
+
+    #[cfg(target_arch = "x86_64")]
+    const __PTHREAD_COND_SIZE__: usize = 40 - 4;
+
+    #[cfg(target_arch = "x86_64")]
+    const __PTHREAD_RWLOCK_SIZE__: usize = 64 - 4;
+
+    const _PTHREAD_MUTEX_MAGIC_INIT: libc::c_uint = 0x33330003;
+    const _PTHREAD_COND_MAGIC_INIT: libc::c_uint = 0x55550005;
+    const _PTHREAD_RWLOCK_MAGIC_INIT: libc::c_uint = 0x99990009;
+
+    // note the actual structs are smaller
+
+    #[repr(C, packed)]
+    pub struct pthread_mutex_t {
+        __magic: libc::c_uint,
+        __opaque: [u8; __PTHREAD_MUTEX_SIZE__],
+    }
+    #[repr(C, packed)]
+    pub struct pthread_mutexattr_t {
+        __opaque: [u8; 16],
+    }
+    #[repr(C, packed)]
+    pub struct pthread_cond_t {
+        __magic: libc::c_uint,
+        __opaque: [u8; __PTHREAD_COND_SIZE__],
+    }
+    #[repr(C, packed)]
```

Comment:
Oh dear, you are right, this is completely wrong. I added `packed` to avoid padding, which is indeed not needed. But as a side effect it completely messed up alignment -  even without `packed` the alignment was wrong for some types, I pushed a fix.